app/controllers/api/v1/activity_sessions_controller.rb

```diff
@@ -84,15 +84,15 @@ def activity_session_params
                   :completed_at,
                   :activity_uid,
                   :anonymous,
-                  concept_tag_results_attributes:  concept_tag_keys)
+                  concept_results_attributes:  concept_tag_keys)
       .merge(data: @data).reject {|k,v| v.nil? }
   end
 
   # Grab a list of all the arbitrarily-named keys that are provided in the concept tag results payload.
   # Returns a list of symbols, e.g. [:student_input, :wpm]
-  def concept_tag_result_allowed_keys
-    if params[:concept_tag_results_attributes]
-      params[:concept_tag_results_attributes].reduce [] do |acc, hash|
+  def concept_result_allowed_keys
```

Comment:
Ok gotcha. I would be in favor of not allowing the hash to be freeform - just specifically permitting those keys which we know QG will be passing back. When we integrate more apps, we can add their keys (with perhaps a switch statement depending on which activity type the data is coming from). I understand that this might be less abstract and force the lms to know about the data structure of the other app, but since this is the data that the LMS is specifically asking for, the LMS is probably going to have to know about its structure somewhere in the LMS codebase in order to utilize that data (ie present to teachers).